impl/src/main/java/io/jsonwebtoken/impl/DefaultJwtParser.java

```diff
@@ -617,11 +616,7 @@ protected JwtSignatureValidator createSignatureValidator(SignatureAlgorithm alg,
 
     @SuppressWarnings("unchecked")
     protected Map<String, ?> readValue(String val) {
-        try {
-            byte[] bytes = val.getBytes(Strings.UTF_8);
-            return deserializer.deserialize(bytes);
-        } catch (DeserializationException e) {
-            throw new MalformedJwtException("Unable to read JSON value: " + val, e);
-        }
+        byte[] bytes = val.getBytes(Strings.UTF_8);
+        return deserializer.deserialize(bytes);
```

Comment:
Does this actually compile? `deserialize` throws `DeserializationException`, which is a checked exception.

Reply:
`DeserializationException` is unchecked, everything compiles and tests run ;)